### PR TITLE
RepeatedCascade: one model, batched parameter values, one trace

### LINKED ISSUE #176: Add RepeatedCascade: one model, batched parameter values, one trace
## What to build

A public `RepeatedCascade` model: one member model plus a mapping from dotted parameter names to arrays whose leading axis is the repeat axis, with every other parameter shared across the repeats. It builds all sections under a single vmap and reduces them with the `pmrf.rf` functions extracted in #175.

Today a cascade of ten microstrip lines with independent widths is built by a Python list comprehension, which traces once per member and gives every member its own name set. `RepeatedCascade` replaces that with one trace and one name set, with the repeat index as a batch axis.

The section count is static by construction: it is the leading axis length of the supplied value arrays, so it is available at trace time without any rule that reads dynamic data.

Note that "batch" in ParamRF already means the parameter batch dimension (`prf.batch_axes`, `prf.sweep`). A `RepeatedCascade` can itself carry that dimension — sweeping a set of widths — so the docstring must be explicit that the repeat axis is not the parameter batch axis. This is why the class is not called `BatchedCascade`. See ADR-0004.

## Acceptance criteria

- [ ] `RepeatedCascade` is a public model, exported from `pmrf.models`, taking a member model and a mapping from dotted parameter names to arrays with a leading repeat axis.
- [ ] Parameters not named in the mapping are shared across all repeats.
- [ ] The repeat count is static; a mapping whose arrays disagree on their leading axis length raises at construction.
- [ ] Sections are built under a single vmap, not a Python loop, and reduced through the `pmrf.rf` functions.
- [ ] S-parameters match a hand-built `Cascade` of the equivalent independently-parametrised members, to numerical tolerance.
- [ ] The model still works when its own parameters carry a sweep batch dimension, and the docstring distinguishes the two axes.
- [ ] Report on #45 how much of that issue this closes and what, if anything, remains.

## Blocked by

- #175

### PR DESCRIPTION
**Pull request by gvcallen**
Adds a public `RepeatedCascade`: one member model plus a mapping from dotted parameter names to arrays whose leading axis is the repeat axis; every other parameter is shared. Sections are built under a single `jax.vmap` and reduced through the `pmrf.rf` functions from #175 (unmodified). Repeat count is static from the leading axis length; mismatched lengths, scalars, an empty mapping, an unknown name and an odd-port member all raise at construction.

S-parameters match a hand-built `Cascade` of equivalent independently-parametrised members to ~1e-15 (S) and 1e-13 (ABCD), in both `method='a'` and `'s'`. The sweep batch axis is tested both over a shared parameter and over whole tapers, and the docstring has a "Repeat axis versus batch axis" section citing ADR-0004. Reported on #45.

New tests: `tests/test_models/test_repeated_cascade.py`, 19 tests. Full suite: 880 passed, 8 failed, 1 skipped — the 8 are the pre-existing sampling/inference `NameError`s also present on the base.

Three points for review:

- **`method` defaults to `'a'`**, unlike `Cascade`'s `'s'`, following ADR-0004's "ABCD domain by default"; `Cascade`'s own default is untouched.
- **A repeated parameter is fixed on the stored member at construction**, since the mapping supplies its value per repeat and leaving it free would hand an optimiser a parameter that moves nothing — the bug reported in #45. Mirrors ADR-0004's rule for `ProfiledLine` targets.
- **`error_if` under `vmap`**: `prf.update` in declared space runs a constraint check via `pmrf.utils.debug.error_if`, whose `lax.cond` degrades to a select under `vmap` and prints the debug branch unconditionally. `_sections` therefore validates once outside the vmap against the whole repeat axis and writes raw values inside it. Real violations still raise. This is a latent hazard in the existing utility, worked around rather than fixed here — worth a follow-up ticket.

Closes #176

Stacked on #181; merge that first

### CODE REVIEW

File: pmrf/models/composite/interconnected/cascade.py [hunk 1]
```diff
@@ -120,109 +122,274 @@ def _evaluate_abcd(self, freq: Frequency) -> jnp.ndarray:
...
-        # space. A diagonal nudge turns that unobservable direction into a very
-        # large, epsilon-dependent solution. The Moore-Penrose inverse instead
-        # eliminates only the observable subspace and is identical to an inverse
-        # for full-rank connection matrices.
-        rtol = max(self.eps, np.finfo(Smat_A.real.dtype).eps * 10)
-        X = jnp.linalg.pinv(M, rtol=rtol)
-        Y = jnp.linalg.pinv(N_mat, rtol=rtol)
+class RepeatedCascade(Model):
+    r"""
+    A cascade of many copies of one model, built under a single vmap.
```
Comment (gvcallen) [inline pmrf/models/composite/interconnected/cascade.py:176]: these docs are very verbose - please trim them

**Review by gvcallen — COMMENTED** [review]
accepted besides docs

File: pmrf/models/composite/interconnected/cascade.py (hunk 1 above)
Comment (gvcallen) [reply pmrf/models/composite/interconnected/cascade.py:176]: Trimmed in b926f2d. The class docstring goes from 106 to 79 lines: dropped the "batched counterpart of Cascade" paragraph that restated the opener, the second worked example of the per-section name set, the Parameters prose that restated the signature, and the import boilerplate in Examples; the `_sections` docstring is halved where it repeated the class. Kept the "Repeat axis versus batch axis" section with its ADR-0004 citation and the Mathematical Formulation. Comments only, no behaviour change; suite unchanged.
